contracts/cw721-base/helpers.ts

```diff
@@ -1,107 +1,100 @@
+ *
+ * Instantiate contract example:
+ *   const initMsg = {
+ *     name: "Potato Coin",
+ *     symbol: "TATER",
+ *     minter: addr
+ *   };
+ *   const instance = await contract.instantiate(addr, codeId, initMsg, 'Potato Coin!');
  * If you want to use this code inside an app, you will need several imports from https://github.com/CosmWasm/cosmjs
  */
 
-const path = require("path");
-
 interface Options {
   readonly httpUrl: string
   readonly networkId: string
   readonly feeToken: string
-  readonly gasPrice: GasPrice
   readonly bech32prefix: string
   readonly hdPath: readonly Slip10RawIndex[]
-  readonly faucetToken: string
   readonly faucetUrl?: string
-  readonly defaultKeyFile: string
-  readonly gasLimits: Partial<GasLimits<CosmWasmFeeTable>>
-}
-
-const defaultOptions: Options = {
-  httpUrl: 'https://rest.cosmwasm.hub.hackatom.dev',
-  networkId: 'hackatom-wasm',
-  feeToken: 'ucosm',
-  gasPrice: GasPrice.fromString("0.025ucosm"),
-  bech32prefix: 'cosmos',
-  faucetToken: 'COSM',
-  faucetUrl: 'https://faucet.cosmwasm.hub.hackatom.dev/credit',
-  hdPath: makeCosmoshubPath(0),
-  defaultKeyFile: path.join(process.env.HOME, ".hackatom.key"),
-  gasLimits: {
-    upload: 1500000,
-    init: 600000,
-    register: 800000,
-    transfer: 80000,
-  },
+  readonly defaultKeyFile: string,
+  readonly fees: {
+    upload: StdFee,
+    init: StdFee,
+    exec: StdFee
+  }
 }
 
-const localnetOptions: Options = {
-  httpUrl: "http://localhost:1317",
-  networkId: 'localnet',
-  feeToken: 'ucosm',
-  gasPrice: GasPrice.fromString("0.025ucosm"),
-  bech32prefix: 'cosmos',
+const pebblenetGasPrice = GasPrice.fromString("0.01upebble");
+const pebblenetOptions: Options = {
+  httpUrl: 'https://rpc.pebblenet.cosmwasm.com',
+  networkId: 'pebblenet-1',
+  bech32prefix: 'wasm',
+  feeToken: 'upebble',
+  faucetUrl: 'https://faucet.pebblenet.cosmwasm.com/credit',
   hdPath: makeCosmoshubPath(0),
-  faucetToken: "SHELL",
-  faucetUrl: "http://localhost",
-  defaultKeyFile: path.join(process.env.HOME, "localnet.key"),
-  gasLimits: {
-    upload: 1500000,
-    init: 600000,
-    register: 800000,
-    transfer: 80000,
+  defaultKeyFile: path.join(process.env.HOME, ".pebblenet.key"),
+  fees: {
+    upload: calculateFee(1500000, pebblenetGasPrice),
+    init: calculateFee(500000, pebblenetGasPrice),
+    exec: calculateFee(200000, pebblenetGasPrice),
   },
 }
 
 interface Network {
-  setup: (password: string, filename?: string) => Promise<SigningCosmWasmClient>
+  setup: (password: string, filename?: string) => Promise<[string, SigningCosmWasmClient]>
   recoverMnemonic: (password: string, filename?: string) => Promise<string>
 }
 
 const useOptions = (options: Options): Network => {
 
-  const loadOrCreateWallet = async (options: Options, filename: string, password: string): Promise<Secp256k1HdWallet> => {
+  const loadOrCreateWallet = async (options: Options, filename: string, password: string): Promise<DirectSecp256k1HdWallet> => {
```

Comment:
IMO this helper file is doing too much. Would rather have the helpers take signing client and then get rid of all this stuff related to wallet creation, faucets, and config.

There is already the `@cosmicdapps/logic` package which handles all that. Would rather these helpers _only_ handle interacting with the chain when given a `CosmWasmClient` or `SigningCosmWasm` client.

Maybe delete all this stuff? Not sure if others have strong opinions here.

On a side note, been brain storming with `ebaker` about how to auto-generate this stuff from the contract schema.

Reply:
I agree with you. We are doing a lot of copy paste. It would be better to collect boilerplate somehwere.

Reply:
maybe a base helper in `cw-plus` if it needs collecting somewhere for the purposes of documentation or tutorials?

Reply:
`base-helper.ts` I like this idea